inet/vibe/inet/url.d

```diff
@@ -419,6 +420,94 @@ struct URL {
 		return cast(NativePath)this.path;
 	}
 
+	/// Decode percent encoded triplets for unreserved or convert to uppercase
+	private string normalize_percent_encoding(const string input)
```

Comment:
In general, a function should not allocate if it doesn't have it. So if it's taking a `string`, it should return that same `string` if it has nothing to normalize. If the function decides to always allocate, it doesn't need to take a `string`, it should take a `scope const(char)`, as `string` implies `immutable`, but the function does not require immutability.

Reply:
Normalization process always allocates without checking need for it (i.e. schema and host `toLower` case); I think we can do it for this one also.